lib/virtual_host_service_worker/amqp_dispatcher.rb

```diff
@@ -1,32 +1,38 @@
 module VirtualHostServiceWorker
-  
+
   class AmqpDispatcher
-    
+
     ##
     # dispatches amqp-messages to method calls
     #
     def self.dispatch(payload)
       DaemonKit.logger.info("AMQP message received")
-      if payload['ssl_certificate'] and payload['ssl_ca_certificate'] and payload['ssl_key']
-        DaemonKit.logger.info("added a new vhost")
-        
-        if APP_CONFIG['use_haproxy'] == true
-          VirtualHostServiceWorker::HaproxyVHostWriter.setup_v_host(payload)
-        else
-          VirtualHostServiceWorker::NginxVHostWriter.setup_v_host(payload)
-        end
+      begin
+        if payload['ssl_certificate'] and payload['ssl_ca_certificate'] and payload['ssl_key']
+          DaemonKit.logger.info("adding a new vhost: #{payload['server_name']}")
 
-      elsif payload['action'] == 'delete'
-        DaemonKit.logger.info("deleted a vhost")
+          if APP_CONFIG['use_haproxy'] == true
+            VirtualHostServiceWorker::HaproxyVHostWriter.setup_v_host(payload)
+            DaemonKit.logger.info("Succesfully added vhost #{payload['server_name']}")
+          else
+            VirtualHostServiceWorker::NginxVHostWriter.setup_v_host(payload)
+            DaemonKit.logger.info("Succesfully added vhost #{payload['server_name']}")
```

Comment:
DRY: Place the logging after the if/else. Alternatively add to the logger which branch you are currently in. (e.g. use_haproxy=true)